File: problems/1566/solution.cpp
class Solution {
public:
    bool containsPattern(vector<int>& arr, int m, int k) {
        for (int i = m, count = 0; i < arr.size(); i++) {
            if (arr[i] != arr[i - m]) {
                count = 0;
            } else if (++count == m * (k - 1)) {
                return true;
            }
        }
        
        return false;
    }
};

File: problems/1566/solution.go
package main
func containsPattern(arr []int, m int, k int) bool {
    for i, count := m, 0; i < len(arr); i++ {
        if arr[i] != arr[i - m] {
            count = 0
        } else {
            count++
            
            if count == m * (k - 1) {
                return true
            }
        }
    }
    
    return false
}